src/vs/workbench/common/editor/editorGroupModel.ts

```diff
@@ -692,11 +692,11 @@ export class EditorGroupModel extends Disposable implements IEditorGroupModel {
 
 	private doSetActive(editor: EditorInput, editorIndex: number): void {
 		if (this.matches(this.active, editor)) {
-			this.selection = [editor];
+			this.selection = [editor]; // TODO this potentially changes selection without a related event. and is it expected to change the selection?
 			return; // already active
 		}
 
-		this.active = editor;
+		this.active = editor; // TODO this potentially changes selection without a related event. and is it expected to change the selection?
```

Comment:
@benibenj this changes selection without event

Reply:
This is because `doSetActive` is only called by `doSetSelection` which handles the event. We could just have one method that combines `doSetActive` and `doSetSelection`

Reply:
I removed `doSetActive` and only use `doSetSelection`. This makes it very clear what happens and ensures we change the selection in the model atomically.